### PR TITLE
[#465] Find record references

### LINKED ISSUE #465: Find references for records
Find references for records. It should be possible to find references using:
- The record definition (e.g. `-record(record_name, {foo, bar}).`)
- Record usage (e.g. `#record_name{}`)
- Record field usage (e.g. `#record_name.foo`)

### PR DESCRIPTION
**Pull request by jfacorro**
### Description

Implements the possibility to find references for a record. 

Fixes #465

### CODE REVIEW

**Comment by jfacorro** [comment]
I'm fixing the issues.

**Comment by onno-vos-dev** [comment]
Strangely enough I had to `rm -rf` my `~/.cache/erlang_ls` to get the references found, not sure if it makes sense to add this somewhere because this is a lovely addition! <3

**Comment by robertoaloi** [comment]
@jfacorro Did you bump the schema version?

**Comment by jfacorro** [comment]
> @jfacorro Did you bump the schema version?

@robertoaloi No, but the schema didn't change. Is that something that we are doing?

File: src/els_references_provider.erl [hunk 1]
```diff
@@ -56,6 +56,21 @@ find_references(Uri, #{ kind := Kind
     {ok, Refs} ->
       [location(U, R) || #{uri := U, range := R} <- Refs]
   end;
+find_references(_Uri, #{kind := Kind, id := Name})
+  when Kind =:= record_expr; Kind =:= record ->
+  case els_dt_references:find_by_id({record, Name}) of
```
Comment (robertoaloi) [inline src/els_references_provider.erl]: Nit: we could refactor this into a function and avoid the duplication.

File: src/els_indexer.erl [hunk 2]
```diff
@@ -90,6 +90,8 @@ do_index(#{uri := Uri, id := Id, kind := Kind} = Document) ->
   %% References
   POIs  = els_dt_document:pois(Document, [ application
                                          , implicit_fun
+                                         , record_expr
```
Comment (robertoaloi) [inline src/els_indexer.erl]: Nit: keep alphabetical ordering

**Comment by robertoaloi** [comment]
Actually, the schema version is not bumped yet. I will do it.

**Pull request merged**